### PR TITLE
search contexts: single page redesign

### LINKED ISSUE #20616: search contexts: update profile page with new design
Designs: https://www.figma.com/file/4Fy9rURbfF2bsl4BvYunUO/RFC-261%3A-Search-Contexts?node-id=1318%3A17488

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #20616

### CODE REVIEW

File: client/web/src/searchContexts/SearchContextPage.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
+.search-context-page {
```
Comment (limitedmage) [inline client/web/src/searchContexts/SearchContextPage.scss:1]: We should use CSS Modules instead for new code

File: client/web/src/searchContexts/SearchContextPage.story.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { WebStory } from '../components/WebStory'
+
+import { SearchContextPage } from './SearchContextPage'
+
+const { add } = storiesOf('web/searchContexts/SearchContextPage', module)
+    .addParameters({
+        chromatic: { viewports: [1200] },
+    })
+    .addDecorator(story => (
+        <div className="p-3 container web-content" style={{ position: 'static' }}>
```
Comment (limitedmage) [inline client/web/src/searchContexts/SearchContextPage.story.tsx:17]: nit: Why is `position: static` needed? This should be the default

File: client/web/src/searchContexts/SearchContextPage.tsx [hunk 3]
```diff
@@ -23,38 +32,83 @@ export const SearchContextPage: React.FunctionComponent<SearchContextPageProps>
...
+                            <PageTitle title={searchContextOrError.spec} />
+                            <div className="mb-2 d-flex align-items-center">
+                                <Link to="/contexts">
+                                    <ChevronLeftIcon className="search-context-page__back-icon" />
+                                </Link>
+                                <h1 className="mb-0">{searchContextOrError.spec}</h1>
+                                {!searchContextOrError.public && (
+                                    <div className="badge badge-pill badge-secondary ml-1">Private</div>
+                                )}
+                            </div>
```
Comment (limitedmage) [inline client/web/src/searchContexts/SearchContextPage.tsx:55]: Can we use the new PageHeader component added here: https://github.com/sourcegraph/sourcegraph/pull/20647 ?

File: client/web/src/searchContexts/SearchContextPage.tsx [hunk 4]
```diff
@@ -23,38 +32,83 @@ export const SearchContextPage: React.FunctionComponent<SearchContextPageProps>
...
+                                        </div>
+                                    ))}
+                                </>
+                            )}
+                        </>
+                    )}
+                    {isErrorLike(searchContextOrError) && (
+                        <div className="alert alert-danger">
+                            Error while loading the search context: <strong>{searchContextOrError.message}</strong>
+                        </div>
```
Comment (limitedmage) [inline client/web/src/searchContexts/SearchContextPage.tsx:109]: suggestion: add storybook test for this error state (and loading state too)

**Review by quinnkeast — APPROVED** [review]

**Review by limitedmage — APPROVED** [review]

**Pull request merged**